src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs

```diff
@@ -116,6 +120,89 @@ public void InitializeService(ServiceHost serviceHost)
             await HandleGetTableOrViewRequest(metadataParams, "view", requestContext);
         }
 
+        /// <summary>
+        /// Handles the event for generating server contextualization scripts. The generated scripts are create
+        /// scripts for database objects like tables and views.
+        /// </summary>
+        internal static async Task HandleGenerateServerContextualizationNotification(GenerateServerContextualizationParams contextualizationParams,
+            EventContext eventContext)
+        {
+            MetadataService.ConnectionServiceInstance.TryFindConnection(contextualizationParams.OwnerUri, out ConnectionInfo connectionInfo);
+
+            if (connectionInfo != null)
+            {
+                using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connectionInfo, "metadata"))
+                {
+                    try
+                    {
+                        // If scripts have been generated within the last 30 days then there isn't a need to go through the process
+                        // of generating scripts again.
+                        if (MetadataScriptTempFileStream.IsScriptTempFileUpdateNeeded(connectionInfo.ConnectionDetails.ServerName))
+                        {
+                            return;
+                        }
+
+                        var scripts = await SmoScripterFactory.GenerateAllServerTableScripts(sqlConn);
```

Comment:
We shouldn't block the message thread while generating scripts since that could take a while. Please sync up with @cheenamalhotra on how to handle this better